### PR TITLE
[docs] [integrations] added RabbitMQ and Apache Beam

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /preview/integrations/rabbitmq/

### CODE REVIEW

File: docs/content/preview/integrations/apache-beam.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+## Prerequisite
+
+To use Apache Beam, ensure that you have YugabyteDB up and running. Download and install YugabyteDB by following the steps in [Quick start](../../quick-start/).
+
+## Setup
+
+To run Apache Beam with YugabyteDB, do the following:
+
+1. Create a folder "apache-beam-test" as follows:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-beam.md:23]:
```suggestion
1. Create a folder `apache-beam-test` as follows:
```

File: docs/content/preview/integrations/apache-beam.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+   source myenv/bin/activate
+   ```
+
+1. Install the latest Apache Beam Python SDK from Pypi using the following command:
+
+    ```python
+    pip install beam-nuggets
+    ```
+
+1. Create a python file, "democode.py" and add the following code to it:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-beam.md:42]:
```suggestion
1. Create a python file, `democode.py` and add the following code to it:
```

File: docs/content/preview/integrations/apache-beam.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        months = p | "Reading month records" >> beam.Create(records)
+        months | 'Writing to DB table' >> relational_db.Write(
+            source_config=source_config,
+            table_config=table_config
+        )
+    if __name__ == "__main__":
+        print('demo code ran successful')
+    ```
+
+1. Run "democode.py" as follows:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-beam.md:79]:
```suggestion
1. Run `democode.py` as follows:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+description: Use RabbitMQ to work with distributed SQL databases in YugabyteDB.
+menu:
+  preview_integrations:
+    identifier: rabbitmq
+    parent: integrations
+    weight: 571
+type: docs
+---
+
+[RabbitMQ](https://www.rabbitmq.com/) is one of the most popular open source message brokers. RabbitMQ is lightweight and easy to deploy on premises and in the cloud. It supports multiple messaging protocols and [streamin](https://www.rabbitmq.com/streams.html). RabbitMQ can be deployed in distributed and federated configurations to meet high-scale, high-availability requirements.
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:14]:
```suggestion
[RabbitMQ](https://www.rabbitmq.com/) is one of the most popular open source message brokers. RabbitMQ is lightweight and easy to deploy on-premises and in the cloud. It supports multiple messaging protocols and [streaming](https://www.rabbitmq.com/streams.html). RabbitMQ can be deployed in distributed and federated configurations to meet high-scale, high-availability requirements.
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+          <properties>
+            <skipITs>true</skipITs>
+            <maven.test.skip>true</maven.test.skip>
+          </properties>
+        </profile>
+      </profiles>
+    </project>
+    ```
+
+1. Create a "class App.java" file in the "src" folder, and add the following code to it. This is the entry point for your application to run:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:245]:
```suggestion
1. Create a `App.java` class file in the `src` folder, and add the following code to it. This is the entry point for your application to run:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 6]
```diff
@@ -0,0 +1,474 @@
...
+          engineRunner.run();
+        } catch (Exception e) {
+          System.out.println("Exception while trying to run the engine: " + e);
+          System.exit(-1);
+        }
+      }
+    }
+    ```
+
+1. Create a configuration file "CmdLineOpts.java" in the "src" folder, where you can store the connection properties to connect to YugabyteDB. Add the following code to it. Set the "streamId" to the stream ID generated in step 1 :
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:263]:
```suggestion
1. Create a configuration file `CmdLineOpts.java` in the `src` folder, where you can store the connection properties to connect to YugabyteDB. Add the following code to it. Set the "streamId" to the stream ID generated in step 1:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+        props.setProperty("database.password", databasePassword);
+        props.setProperty("database.dbname", databaseName);
+        props.setProperty("database.server.name", "dbserver1");
+        props.setProperty("snapshot.mode", snapshotMode);
+        return props;
+      }
+    }
+    ```
+
+1. Create a "Send.java" class in the "src" folder. This creates a channel between the producer and the queue and then publishes the message to the queue:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:336]:
```suggestion
1. Create a `Send.java` class in the `src` folder. This creates a channel between the producer and the queue and then publishes the message to the queue:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 8]
```diff
@@ -0,0 +1,474 @@
...
+                channel.queueDeclare(QUEUE_NAME, false, false, false, null);
+                String message = msg;
+                channel.basicPublish("", QUEUE_NAME, null, message.getBytes(StandardCharsets.UTF_8));
+                System.out.println(" [x] Sent '" + message + "'");
+            }
+        }
+    }
+    ```
+
+1. Create an "EngineRunner.java" class in the "src" folder, which starts the debezium engine and captures the changes on the server side as records. It then calls the "Send.java" class's run API with the record that it received to publish it to the queue.
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:361]:
```suggestion
1. Create an `EngineRunner.java` class in the `src` folder, which starts the Debezium engine and captures the changes on the server side as records. It then calls the `Send.java` class run API with the record that it received to publish it to the queue.
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 9]
```diff
@@ -0,0 +1,474 @@
...
+          executor.execute(engine);
+        } catch (Exception e) {
+          System.out.println(e);
+          throw e;
+        }
+      }
+    }
+    ```
+
+1. Finally, create a class "Recv.java" in the "src" folder which consumes the records from the queue and outputs to the console:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:411]:
```suggestion
1. Finally, create a class `Recv.java` in the `src` folder which consumes the records from the queue and outputs to the console:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 10]
```diff
@@ -0,0 +1,474 @@
...
+            DeliverCallback deliverCallback = (consumerTag, delivery) -> {
+                String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
+                System.out.println(" [x] Received '" + message + "'");
+            };
+            channel.basicConsume(QUEUE_NAME, true, deliverCallback, consumerTag -> { });
+        }
+    }
+    ```
+
+1. Run the "Recv.java" class for the receiver to start listening. You can see output similar to the following:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:439]:
```suggestion
1. Run the `Recv.java` class for the receiver to start listening. You can see output similar to the following:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 11]
```diff
@@ -0,0 +1,474 @@
...
+
+    ```output
+    [*] Waiting for messages. To exit press CTRL+C
+    ```
+
+1. From a new terminal run the "App.java" class.
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:445]:
```suggestion
1. From a new terminal run the `App.java` class.
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 12]
```diff
@@ -0,0 +1,474 @@
...
+1. After both the applications are running, open the ysqlsh shell and insert a row into the test table as follows:
+
+    ```sql
+    INSERT into test values (1, Jake, 20);
+    ```
+
+1. From the terminal where "App.java" runs, you can see a log output similar to the following:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:452]:
```suggestion
1. From the terminal where `App.java` runs, you can see a log output similar to the following:
```

File: docs/content/preview/integrations/rabbitmq.md [hunk 13]
```diff
@@ -0,0 +1,474 @@
...
+    [x] Sent 'EmbeddedEngineChangeEvent [key={"schema":{"type":"struct","fields":[{"type":"struct","fields":[{"type":"int32","optional":false,"field":"value"},{"type":"boolean","optional":false,"field":"set"}],"optional":false,"name":"id","field":"id"}],"optional":false,"name":"dbserver1.public.test.Key"},"payload":{"id":{"value":8,"set":true}}}
+    .
+    .
+    .
+    .
+    .
+    valueSchema=Schema{dbserver1.public.test.Envelope:STRUCT}, timestamp=null, headers=ConnectHeaders(headers=)}]
+    ```
+
+1. From the terminal where "Recv.java" runs, you can see a log output similar to the following:
```
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:464]:
```suggestion
1. From the terminal where `Recv.java` runs, you can see a log output similar to the following:
```

**Review by ddhodge — COMMENTED** [review]
misc comments

File: docs/content/preview/integrations/rabbitmq.md [hunk 14]
```diff
@@ -0,0 +1,474 @@
...
+        CmdLineOpts configuration = new CmdLineOpts();
+        configuration.initialize(commandLine);
+        return configuration;
+      }
+      private void initialize(CommandLine commandLine) {
+          masterAddresses = "127.0.0.1:7100";
+          String[] nodes = masterAddresses.split(",");
+          hostname = nodes[0].split(":")[0];
+
+          streamId = "962c322b60af4b9abd0b195fc6e3f045";
```
Comment (Sfurti-yb) [inline docs/content/preview/integrations/rabbitmq.md:313]: Please change this stream id to the output of the stream id generated in step 1. So to this 
```
CDC Stream ID: 31dd4440caca46038ba4a9365bb89d36
```

**Review by Sfurti-yb — APPROVED** [review]
Minor comment. Other than that LGTM.

File: docs/content/preview/integrations/rabbitmq.md (hunk 14 above)
Comment (ddhodge) [inline docs/content/preview/integrations/rabbitmq.md:313]:
```suggestion
          streamId = "31dd4440caca46038ba4a9365bb89d36";
```

**Comment by aishwarya24** [comment]
Thanks @Sfurti-yb and @ddhodge

**Pull request merged**